Controller.cpp

```diff
@@ -523,6 +591,28 @@ namespace Procon {
 		return sendCommand(0x10, buf);
 	}
 
+	Controller::exchangeArray Controller::getSpiData(uint32_t offset, const uint8_t readLen)
+	{
+		std::array<uint8_t, 0x100> buf;
+		while (true)
+		{
+			buf.fill(0);
+			*(uint32_t*)(buf.data()) = offset;
+			*(uint8_t*)(buf.data() + 4) = readLen;
+			auto res = sendSubcommand(0x01, 0x10, buf);
+			//if (res && (*(uint16_t*)(res.value().data() + 0xD) == 0x1090) && (*(uint32_t*)(res.value().data() + 0xF) == offset)) // For bluetooth
+			if (res && (*(uint16_t*)(res.value().data() + 0x17) == 0x1090) && (*(uint32_t*)(res.value().data() + 0x19) == offset))
```

Comment:
I know they're long, but use C++ casts.

Reply:
I already answered this a few comments above.